### PR TITLE
MODINREACH-420 Updating dependencies for Quesnelia

### PR DESCRIPTION
**Pull request by Vignesh-kalyanasundaram**
## Purpose
The purpose of the PR is to implement MODINREACH-420

## Approach
When the spring boot starter parent and folio spring base was updated, we faced and fixed the following issues.

1)Caused by: java.lang.IllegalArgumentException: LoggerFactory is not a Logback LoggerContext but Logback is on the classpath. - This is fixed after the exclusion of logback-classic artifact from spring-kafka-test artifact.

2)Caused by: org.springframework.context.annotation.ConflictingBeanDefinitionException - Annotation-specified bean name 'folioTenantController' for bean class [org.folio.innreach.controller.base.BaseControllerTest$TestTenantController] conflicts with existing, non-compatible bean definition of same name and class [org.folio.innreach.domain.listener.base.BaseKafkaApiTest$TestTenantController] - This is fixed by removing the duplicate restcontroller from the test classes

3)After Openapi version upgradation to 7.3.0, it generates instanceidentifiersinner class instead of instanceindentifiers class. It looks like the entire identifier object is declared inside instance.json file so it is creating such kind of class. This is breaking in the existing class which are refering instanceidentifiers. So instanceIdentifiers.json file is created and referred from instance.json file. same approach followed for contributors.json as well

#### TODOS and Open Questions

- [ ] Check logging

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by gurleenkaurbp — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -23,32 +23,30 @@
...
     <mapstruct.version>1.5.3.Final</mapstruct.version>
     <marc4j.version>2.9.2</marc4j.version>
     <coffee-boots.version>3.0.0</coffee-boots.version>
-    <snakeyaml.version>1.33</snakeyaml.version>
 
     <!-- test dependencies -->
-    <testcontainers.version>1.17.6</testcontainers.version>
-    <wiremock.version>2.35.0</wiremock.version>
+    <testcontainers.version>1.19.0</testcontainers.version>
+    <wiremock-standalone.version>3.0.1</wiremock-standalone.version>
```
Comment (julianladisch) [inline pom.xml]: Please fix CVE-2023-41329 vulnerability by upgrading to latest `org.wiremock:wiremock-standalone` version.

Comment (Vignesh-kalyanasundaram) [reply pom.xml]: Hi @julianladisch , I tried to upgrade the latest version 3.4.2 as well as the patched version 3.0.3 of org.wiremock:wiremock-standalone. but none of them are working. It is failing with below error.
 **Could not find artifact com.github.tomakehurst:wiremock-standalone:jar:3.0.3 in folio-nexus (https://repository.folio.org/repository/maven-public)**

Comment (Vignesh-kalyanasundaram) [reply pom.xml]: @julianladisch As suggested by you, I changed the group id from com.github.tomakehurst to org.wiremock. It is working now. Thanks.

File: pom.xml [hunk 2]
```diff
@@ -23,32 +23,30 @@
   <properties>
     <!-- runtime dependencies -->
     <folio-service-tools.version>3.1.0</folio-service-tools.version>
-    <folio-spring-base.version>7.2.0</folio-spring-base.version>
-    <folio-spring-system-user.version>7.2.1</folio-spring-system-user.version>
-    <spring-cloud-starter-openfeign.version>4.0.3</spring-cloud-starter-openfeign.version>
+    <folio-spring-base.version>8.1.0</folio-spring-base.version>
+    <folio-spring-system-user.version>8.1.0</folio-spring-system-user.version>
```
Comment (julianladisch) [inline pom.xml]: Almost always this is the same version, so these two might be merged into a single value like `<folio-spring.version>` (or `folio-spring-support.version>`.

File: pom.xml [hunk 3]
```diff
@@ -23,32 +23,30 @@
...
     <commons-collections4.version>4.4</commons-collections4.version>
     <mapstruct.version>1.5.3.Final</mapstruct.version>
     <marc4j.version>2.9.2</marc4j.version>
     <coffee-boots.version>3.0.0</coffee-boots.version>
-    <snakeyaml.version>1.33</snakeyaml.version>
 
     <!-- test dependencies -->
-    <testcontainers.version>1.17.6</testcontainers.version>
-    <wiremock.version>2.35.0</wiremock.version>
+    <testcontainers.version>1.19.0</testcontainers.version>
```
Comment (julianladisch) [inline pom.xml]: This line can be removed because spring-boot-starter-parent provides a recent `testcontainers.version` .

File: pom.xml (hunk 2 above)
Comment (Vignesh-kalyanasundaram) [reply pom.xml]: Sure, changed to `<folio-spring.version>`

File: pom.xml (hunk 3 above)
Comment (Vignesh-kalyanasundaram) [reply pom.xml]: sure, removed the `testcontainers.version `

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Pull request merged**